Source/WebKit/UIProcess/API/Cocoa/NSAttributedString.mm

```diff
@@ -164,6 +184,45 @@ + (void)clearConfiguration
     globalConfiguration() = nil;
 }
 
++ (void)validateEntry:(id)maybeFileURL
+{
+    NSString *errorMessage = nil;
+
+    auto* url = dynamic_objc_cast<NSURL>(maybeFileURL);
+    if (!url)
+        errorMessage = @"The NSDictionary associated with _WKReadAccessFileURLsOption may only contain NSURL objects.";
+    else if (!url.fileURL)
+        errorMessage = @"_WKReadAccessFileURLsOption requires its NSURL objects to be file URLs.";
+
+    if (!errorMessage)
+        return;
+
+    if (self.readOnlyAccessPaths.get().count) {
+        [self.readOnlyAccessPaths removeAllObjects];
```

Comment:
Does this work, or does it just remove the objects from a copy of the array? I think this might need to be `[self.readOnlyAccessPaths setArray:@[]]`.

Reply:
I think this is correct as-is, since `readOnlyAccessPaths` returns a pointer (prior to your edits, a RetainPtr) to the NeverDestroyed NSMutableArray. I'll confirm in the debugger, since the test cases don't exercise this well (if we have no required directories, we can't tell if we have a set of unused directories in the `NSMutableArray` or a truly empty array!)

Reply:
Oh, OK, that’s the function that should be a function and not a method! I didn’t realize that.